config/locals.js

```diff
@@ -33,59 +33,122 @@ function getModelMap(configName, model) {
 }
 
 /**
- * Loads our hashmap app.locals.controllers.campaignBots, storing CampaignBotControllers by bot id.
+ * Upserts given Phoenix campaign to campaign model, saves to app.locals.campaign[campaign.id].
  */
-function loadCampaignBotControllers() {
-  logger.debug('loadCampaignBots');
-  app.locals.controllers.campaignBots = {};
+function loadCampaign(campaign) {
+  logger.debug(`loadCampaign:${campaign.id}`);
+
+  const data = {
+    status: campaign.status,
+    tagline: campaign.tagline,
+    title: campaign.title,
+    rb_confirmed: campaign.reportbackInfo.confirmationMessage,
+    rb_noun: campaign.reportbackInfo.noun,
+    rb_verb: campaign.reportbackInfo.verb,
+  };
+
+  return app.locals.db.campaigns
+    .findOneAndUpdate({ _id: campaign.id }, data, { upsert: true, new: true })
+    .exec()
+    .then((campaignDoc) => {
+      app.locals.campaigns[campaign.id] = campaignDoc;
```

Comment:
I'd add a basic validation for the `campaignDoc` var. Just to make sure that `_id` present.